x-pack/plugins/spaces/server/plugin.ts

```diff
@@ -92,12 +91,16 @@ export class Plugin {
 
   constructor(initializerContext: PluginInitializerContext) {
     this.config$ = initializerContext.config.create<ConfigType>();
+    this.kibanaIndexConfig = initializerContext.config.legacy.globalConfig$;
     this.log = initializerContext.logger.get();
   }
 
   public async start() {}
 
-  public async setup(core: CoreSetup, plugins: PluginsSetup): Promise<SpacesPluginSetup> {
+  public async setup(
+    core: CoreSetup<PluginsSetup>,
```

Comment:
I'm not sure I understand. Should I remove the generic parameter altogether, or am I defining it incorrectly? I based what I have off of my limited understanding of [the interface](https://github.com/elastic/kibana/blob/1bb59af44a5c4a595f9d589e012734952d529a5b/src/core/public/index.ts#L184-L216), where the generic argument appears to be the plugin start contracts that your plugin depends on

Reply:
Sorry for not being clear, so what I meant is that (let's assume we have `PluginB` that depends on `PluginA`):
```typescript
interface DependencyASetup {
  foo: () => void;
}

interface DependencyAStart {
  bar: () => void;
}

interface PluginBSetupDependencies {
  a: DependencyASetup;
}

interface PluginBStartDependencies {
  a: DependencyAStart;
}

---- PLUGIN B-----
public async setup(core: CoreSetup<PluginBSetupDependencies>, deps: PluginBSetupDependencies) {
  const [coreStart, { a }] = await core.getStartServices();
  // because of CoreSetup<PluginBSetupDependencies> the type of `a`
  // will be `DependencyASetup`, but in fact it should be `DependencyAStart`
  // that may and will be different most of the time, so we should use
  // CoreSetup<PluginBStartDependencies> instead.
  // See https://github.com/elastic/kibana/blob/06df2b0db65a97888d47fed024597e627a3290e0/src/core/public/plugins/plugin_context.ts#L111
  a.foo(); ---> run time error
}
```

I guess it works for you right now since both `features` start and setup contracts have `getFeatures` method, but it's just coincidence.

Reply:
OHH I was incorrectly specifying `PluginsSetup` instead of `PluginsStart`. I intended for `PluginsStart` there. Thanks for explaining and clarifying this for me.